python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -40,16 +40,19 @@
     from pyspark.pandas.series import Series  # noqa: F401 (SPARK-34943)
 
 
-def is_valid_operand_for_numeric_arithmetic(operand: Any) -> bool:
+def is_valid_operand_for_numeric_arithmetic(
+    operand: Any,
+    allow_bool_index_ops: bool = True
+) -> bool:
```

Comment:
Shall we make `allow_bool_index_ops` a keyword-only argument?

```py
def is_valid_operand_for_numeric_arithmetic(
    operand: Any,
    *,
    allow_bool_index_ops: bool = True
) -> bool:
```

Reply:
Btw, the argument name should be like `allow_bool` or something since we will want to use it for bool literal as an operand in a following PR.

Reply:
Good idea, `allow_bool` is set as a keyword-only argument.